src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslX509CertificateReader.cs

```diff
@@ -677,6 +688,37 @@ public ICertificatePal CopyWithPrivateKey(ECDiffieHellman privateKey)
             }
         }
 
+        public ICertificatePal CopyWithPrivateKey(MLDsa privateKey)
+        {
+            SafeX509Handle certHandle = Interop.Crypto.X509UpRef(_cert);
+            OpenSslX509CertificateReader duplicate = new OpenSslX509CertificateReader(certHandle);
+            duplicate._mldsaPrivateKey = DuplicatePrivateKey(privateKey);
+
+            return duplicate;
+        }
+
+        private static MLDsa DuplicatePrivateKey(MLDsa key)
+        {
+            MLDsaAlgorithm alg = key.Algorithm;
+            byte[] rented = CryptoPool.Rent(alg.SecretKeySizeInBytes);
+            int written = 0;
+
+            try
```

Comment:
I assume we're suppose to duplicate handle here but we can't because there is no MLDsaOpenSsl yet?

Reply:
Correct.  We'll still need code like this in case someone calls `cert.CopyWithPrivateKey(someOtherImpl)`, but the expected case would be that it just does `_keyHandle = key.DuplicateHandle()`, where `key` is `MLDsaOpenSsl` or the non-public impl type.